### PR TITLE
SW-484_do_not_export_disabled_properties

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

**Review by TobiasGraml11 — APPROVED** [review]
LGTM!

**Review by TobiasGraml11 — CHANGES_REQUESTED** [review]
Please export non-filterable properties as `Property` in the export

File: tests/Struct/FindologicProductTest.php [hunk 1]
```diff
@@ -494,6 +494,71 @@ function (Attribute $attribute) use ($expectedName) {
...
+        $foundProperties = array_filter(
+            $findologicProduct->getProperties(),
+            function (Property $property) {
+                return $property->getKey() === 'blub';
+            }
+        );
+
+        /** @var Property $property */
+        $property = reset($foundProperties);
+        $this->assertEquals('some value', $property->getAllValues()['']); // '' = Empty usergroup.
```
Comment (dblaichinger) [inline tests/Struct/FindologicProductTest.php:559]: Also extract 'some value' as `$expectedPropertyName`

File: tests/Struct/FindologicProductTest.php [hunk 2]
```diff
@@ -494,6 +494,71 @@ function (Attribute $attribute) use ($expectedName) {
...
+        $productEntity = $this->createTestProduct([
+            'properties' => [
+                [
+                    'id' => Uuid::randomHex(),
+                    'name' => 'some value',
+                    'group' => [
+                        'id' => Uuid::randomHex(),
+                        'name' => 'blub',
+                        'filterable' => false
+                    ],
```
Comment (dblaichinger) [inline tests/Struct/FindologicProductTest.php:512]: It would be good to test `'filterable' => true` as well

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: tests/Struct/FindologicProductTest.php (hunk 2 above)
Comment (TheKeymaster) [reply tests/Struct/FindologicProductTest.php:512]: That is the default. We already have many tests that test this behavior.

File: src/Struct/FindologicProduct.php [hunk 3]
```diff
@@ -425,33 +427,97 @@ protected function getAttributeProperties(ProductEntity $productEntity): array
...
+            }
+        }
+
+        return $properties;
+    }
+
+    /**
+     * @return Attribute[]
+     */
+    protected function getAttributePropertyAsAttribute(
```
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Struct/FindologicProduct.php:489]: I thought a bit, but either we have multiple if statements, which makes this entire block really hardly readable, or we have two separate methods. Maybe you have another idea. I have already invested some time thinking about a better solution though.

File: src/Struct/FindologicProduct.php [hunk 4]
```diff
@@ -425,33 +427,97 @@ protected function getAttributeProperties(ProductEntity $productEntity): array
...
+
+            $groupName = $group->getTranslation('name');
+            $optionName = $settingOption->getTranslation('name');
+            if ($groupName && $optionName) {
+                $configAttrib = new Attribute(
+                    Utils::removeSpecialChars($groupName),
+                    [Utils::removeControlCharacters($optionName)]
+                );
+
+                $attributes[] = $configAttrib;
```
Comment (dblaichinger) [inline src/Struct/FindologicProduct.php:520]: Is the `return` missing?

File: src/Struct/FindologicProduct.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (dblaichinger) [reply src/Struct/FindologicProduct.php:489]: Isn't the only difference the instances, either `new Attribute` or `new Property`? Then I suggest to extract those differences in separate methods and call the methods depending on the case

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: src/Struct/FindologicProduct.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (dblaichinger) [reply src/Struct/FindologicProduct.php:489]: As discussed, we leave it for now.

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**